homeassistant/components/zwave_js/climate.py

```diff
@@ -259,10 +260,22 @@ def _set_modes_and_presets(self) -> None:
     def _current_mode_setpoint_enums(self) -> list[ThermostatSetpointType]:
         """Return the list of enums that are relevant to the current thermostat mode."""
         if self._current_mode is None or self._current_mode.value is None:
-            # Thermostat with no support for setting a mode is just a setpoint
-            if self.info.primary_value.property_key is None:
-                return []
-            return [ThermostatSetpointType(int(self.info.primary_value.property_key))]
+            # Thermostat with no support for setting a mode is just a setpoint.
+
+            # Some devices have missing or invalid setpoint types. For backwards
+            # compatibility we unconditionally map these to the Heating setpoint
+            # (including the actual NA setpoint).
+            setpoint_type = ThermostatSetpointType.NA
+            if self.info.primary_value.property_key is not None:
+                with contextlib.suppress(ValueError):
+                    setpoint_type = ThermostatSetpointType(
+                        int(self.info.primary_value.property_key)
+                    )
+
+            if setpoint_type == ThermostatSetpointType.NA:
+                return [ThermostatSetpointType.HEATING]
```

Comment:
If the device gives us invalid setpoints I suggest we return an empty list instead and raise `HomeAssistantError` with a message in `async_set_temperature` if this is the case. The other places already have guards for falsy return value for `_current_mode_setpoint_enums`.

Additionally we can create a repair issue asking the user to re-interview the device as when the device config has changed.